cms/djangoapps/contentstore/tasks.py

```diff
@@ -231,6 +231,29 @@ def update_library_index(library_id, triggered_time_isoformat):
     else:
         LOGGER.debug('Search indexing successful for library %s', library_id)
 
+@shared_task
+@set_code_owner_attribute
+def update_special_exams_and_publish(course_key_str):
+    """
+    Registers special exams for a given course and calls publishing flow.
+
+    on_course_publish expects that the edx-proctoring subsystem has been refreshed
+    before being executed, so both functions are called here synchronously.
+    """
+    from cms.djangoapps.contentstore.proctoring import register_special_exams
+    from openedx.core.djangoapps.credit.signals import on_course_publish
+
+    course_key = CourseKey.from_string(course_key_str)
+    LOGGER.debug('Attempting to register exams for course %s', course_key_str)
+    try:
+        register_special_exams(course_key)
+        LOGGER.debug('Successfully registered exams for course %s', course_key_str)
+    # pylint: disable=broad-except
+    except Exception as exception:
+        LOGGER.exception(exception)
+
+    LOGGER.debug('Publishing course %s', course_key_str)
+    on_course_publish(course_key)
```

Comment:
I've moved this function inside the async task, because `on_course_publish` must be executed after register_special_exams is completed. I considered making it a callback, but the function is not currently set up as a task (so it can't be linked using an [apply_async call](https://docs.celeryproject.org/en/stable/userguide/calling.html#linking-callbacks-errbacks)).